README.md

```diff
@@ -55,9 +84,85 @@ let tableViewMenu = TableViewMenu(title: "Environment",
 Yoshi.setupDebugMenu([tableViewMenu])
 
 // Invoke Yoshi
-Yoshi.showDebugActionSheet()
+Yoshi.show()
+```
+
+Yoshi will take care of managing selections and call back the convenient closure function when a new selection is made.
+
+### Date Selector Menu
+
+Similarly, to present a date selector menu, create a type that conforms to `YoshiDateSelectorMenu` protocol
+
+```swift
+internal final class DateSelector: YoshiDateSelectorMenu {
+
+    var title: String
+    var subtitle: String?
+    var selectedDate: NSDate
+    var didUpdateDate: (dateSelected: NSDate) -> ()
+
+    init(title: String,
+         subtitle: String? = nil,
+         selectedDate: NSDate = NSDate(),
+         didUpdateDate: (NSDate) -> ()) {
+        self.title = title
+        self.subtitle = subtitle
+        self.selectedDate = selectedDate
+        self.didUpdateDate = didUpdateDate
+    }
+
+}
+```
+
+and present it using the same functions.
+
+```swift
+let dateSelectorMenu = DateSelector(title: "Environment Date",
+    subtitle: nil,
+    didUpdateDate: { (dateSelected) in
+      // Do something with the selected date here
+})
+
+Yoshi.setupDebugMenu([dateSelectorMenu])
+Yoshi.show()
+```
+
+### Custom Menu
+
+Yoshi can also be configured to display custom menus which can be used for triggering events or presenting view controllers.
+
+For example, we can invoke [Instabug](https://instabug.com) when a custom menu is selected.
+
+```swift
+private struct CustomMenu: YoshiMenu {
+
+    let title: String
+    let subtitle: String?
+    let completion: () -> ()
+
+    func execute() -> YoshiActionResult {
+        return .AsyncAfterDismissing(completion)
+    }
+
+}
+
+Instabug.startWithToken("abcdefghijklmnopqrstuvwxyz", invocationEvent: .None)
+Instabug.setDefaultInvocationMode(.BugReporter)
+
+let instabugMenu = CustomMenu(title: "Start Instabug",
+    subtitle: nil,
+    completion: {
+        Instabug.invoke()
+    })
+
+Yoshi.setupDebugMenu([instabugMenu])
+Yoshi.show()
 ```
 
+### Invocation Options
+
+In addition to the vanilla invocation option, Yoshi can also be invoked in response to motion or touch events. To do this, simply forward the motion and touch-related `UIResponder` events to their corresponding Yoshi event-handling functions.
```

Comment:
i don't think we need to list out all the options. i added an example so that people can get an idea of how this works.